packages/database/src/utils/array.ts

```diff
@@ -13,3 +13,13 @@ export function quickpop<T>(arr: T[], index: number): T {
   if (arr.length !== index) arr[index] = elem
   return res
 }
+
+/**
+ * async filter promise all
+ */
+export async function filter(arr: string[], callback) {
+  const fail: string = 'abcd' // temp
+  return (await Promise.all(arr.map(
+    async item => (await callback(item)) ? item : fail
+  ))).filter(i => i !== fail)
+}
```

Comment:
Source: https://stackoverflow.com/questions/33355528/filtering-an-array-with-a-function-that-returns-a-promise

Would've wanted to use generics here, but I'm not sure how to instantiate a unique `T` object.